### DISCUSSION TITLE
Backward compatibility issue

### DISCUSSION THREAD

**Discussion by rogerfzr** [gdsfactory/gdsfactory#1815] [Q&A] [answered]
Hi,

I'm a beginner in gdsfactory. A previous member of my group has created a foundry PDK based on an older version of gdsfactory. When I ran his code using the most recent version of gdsfactory, I was not able to import the PDK as a lot of commands from gdsfactory has changed ever since. It becomes very difficult to track every outdated command and updated it one by one. Unfortunately my group member has (sort of) left the group and is not able to provide help. I know gdsfactory is not backward compatible, but I wonder is there a way to run the old code with the newest version of gdsfactory? Of course I can try installing the older version of gdsfactory but it doesn't seem like a long term solution.

Thank you!
Roger

**Comment by joamatab** [comment]
Hi Roger,

You can keep using the old version of gdsfactory, just make sure you pin the version of gdsfactory that works with your pdk 

You can also try to upgrade yourself or contract a python developer to help you with that

**Reply by tvt173** [reply, marked as answer]
Hi Roger,

Yes, unfortunately gdsfactory has gone through many changes over the past few years, and it can sometimes be difficult to upgrade if you haven't paid attention to it for awhile.

Another thing you can consider doing as a short-term solution is export your cells to GDS (using the old version of gdsfactory), and then re-import them into your new design, using the new gdsfactory. That way, you can start fresh with the latest gdsfactory, use your old cells unchanged, and you can slowly port them back once you get more familiar.

I also recommend you use a strong IDE like PyCharm or VSCode that lets you search and replace, and refactor your whole codebase at once. Depending on how old your library is, it may just be a matter of a few search and replace operations to get it in working condition again.

**Reply by rogerfzr** [reply]
Thanks Joaquin and Troy for the prompt reply! Unfortunately, the previous version of gdsfactory that the PDK was based on is 4.4.6, which is very outdated. I tried to replace a few lines with the new lines but soon realized there are too many updates to cover since the v4.4.6. Particularly, some command has been removed entirely and merged into new ones. I think re-importing the old GDS into the new code is a great idea. I will definitely try that.

Thanks!
Roger
